src/Client/PullPayment.php

```diff
@@ -0,0 +1,336 @@
+            [
+                'name' => $paymentName,
+                'amount' => $paymentAmount,
+                'currency' => $paymentCurrency,
+                'period' => $paymentPeriod,
+                'BOLT11Expiration' => $boltExpiration,
+                'startsAt' => $startsAt,
+                'expiresAt' => $expiresAt,
+                'paymentMethods' => $paymentMethods
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    /**
+    * Archive a Pull Payment
+    *
+    * @return true on success, throws exception
+    */
+
+    public function archivePullPayment(
+        string $storeId,
+        string $pullPaymentId
+    ) {
+        $url = $this->getApiUrl() . 'stores/' .
+        urlencode($storeId) . '/' . 'pull-payments/' .
+        urlencode($pullPaymentId);
+
+        $headers = $this->getRequestHeaders();
+        $method = 'DELETE';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return true;
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    /**
+    * Approve a Payout
+    *
+    * @return string $id
+    * @return int $revision
+    * @return string $pullPaymentId
+    * @return string $date
+    * @return string $destination
+    * @return int $amount
+    * @return string $paymentMethod
+    * @return string $cryptoCode
+    * @return int $paymentMethodAmount
+    * @return string $state
+    */
+
+    public function approvePayout(
+        string $storeId,
+        string $payoutId,
+        int $revision,
+        mixed $rateRule
+    ) {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/' . 'payouts/' .
+                    urlencode($payoutId);
+
+        $headers = $this->getRequestHeaders();
+        $method = 'POST';
+
+        $body = json_encode(
+            [
+                'revision' => $revision,
+                'rateRule' => $rateRule,
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    /**
+    * Cancel a Pull Payment
+    *
+    * @return true on success, throws exception
+    */
+
+    public function cancelPayout(
+        string $storeId,
+        string $payoutId
+    ) {
```

Comment:
you can avoid the docblock and add return type hint `: bool` here, see other classes for examples